lib/netutil/tcplistener.go

```diff
@@ -22,10 +24,20 @@ var enableTCP6 = flag.Bool("enableTCP6", false, "Whether to enable IPv6 for list
 // If useProxyProtocol is set to true, then the returned listener accepts TCP connections via proxy protocol.
 // See https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt
 func NewTCPListener(name, addr string, useProxyProtocol bool, tlsConfig *tls.Config) (*TCPListener, error) {
+	var ln net.Listener
 	network := GetTCPNetwork()
-	ln, err := net.Listen(network, addr)
-	if err != nil {
-		return nil, err
+	listeners, err := activation.ListenersWithNames()
```

Comment:
I'd also add a comment like "Check for socket activations on systemd linux systems".